pkg/scheduler/framework/plugins/interpodaffinity/filtering.go

```diff
@@ -622,16 +631,28 @@ func (pl *InterPodAffinity) Filter(ctx context.Context, cycleState *framework.Cy
 	}
 
 	if s, err := pl.satisfiesExistingPodsAntiAffinity(pod, state, nodeInfo); !s || err != nil {
-		return migration.PredicateResultToFrameworkStatus([]predicates.PredicateFailureReason{predicates.ErrPodAffinityNotMatch, predicates.ErrExistingPodsAntiAffinityRulesNotMatch}, err)
+		if err != nil {
+			return framework.NewStatus(framework.Error, err.Error())
+		}
+		return framework.NewStatus(framework.Unschedulable, ErrReasonAffinityNotMatch, ErrReasonExistingAntiAffinityRulesNotMatch)
 	}
 
 	// Now check if <pod> requirements will be satisfied on this node.
 	affinity := pod.Spec.Affinity
 	if affinity == nil || (affinity.PodAffinity == nil && affinity.PodAntiAffinity == nil) {
 		return nil
 	}
-	if status, err := pl.satisfiesPodsAffinityAntiAffinity(pod, state, nodeInfo, affinity); err != nil || status != nil {
-		return migration.PredicateResultToFrameworkStatus([]predicates.PredicateFailureReason{predicates.ErrPodAffinityNotMatch, status}, err)
+	if status, err := pl.satisfiesPodsAffinityAntiAffinity(pod, state, nodeInfo, affinity); err != nil || status != "" {
+		if err != nil {
+			return framework.NewStatus(framework.Error, err.Error())
+		}
+
+		code := framework.Unschedulable
+		if status == ErrReasonAffinityRulesNotMatch {
```

Comment:
I don't quite like doing string comparisons, lets change ```satisfiesPodsAffinityAntiAffinity``` to return two booleans: one for AffinityRulesNotMatch and one for AntiAffinityRulesNotMatch